runscanner/src/main/java/uk/ac/bbsrc/tgac/miso/runscanner/processors/StandardIllumina.java

```diff
@@ -52,6 +58,16 @@ public NotificationDto process(File runDirectory, TimeZone tz) throws IOExceptio
       throw new IOException(e);
     }
 
+    // See if we can figure out the chemistry
+
+    try {
+      Document parameters = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(new File(runDirectory, "runParameters.xml"));
+      dto.setChemistry(Arrays.stream(IlluminaChemistry.values()).filter(chemistry -> chemistry.test(parameters)).findFirst()
+          .orElse(IlluminaChemistry.UNKNOWN));
+    } catch (SAXException | ParserConfigurationException e) {
+      log.error("Failed to parse parameters", e);
+    }
+
     // The Illumina library can't distinguish between a failed run and one that either finished or is still going. Scan the logs, if
     // available to determine if the run failed.
     File rtaLogDir = new File(runDirectory, "/Data/RTALogs");
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Refactor your code to get this URI from a customizable parameter. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1075)